compat/v0/compat_test.go

```diff
@@ -0,0 +1,99 @@
+package v0_test
+
+import (
+	"context"
+	"database/sql"
+	"encoding/base64"
+	"fmt"
+
+	_ "github.com/mattn/go-sqlite3"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/alicebob/miniredis"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/go-redis/redis/v7"
+	"github.com/renproject/darknode/txengine/txenginebindings"
+	"github.com/renproject/id"
+	v0 "github.com/renproject/lightnode/compat/v0"
+	"github.com/renproject/lightnode/db"
+	"github.com/renproject/lightnode/testutils"
+	"github.com/renproject/multichain"
+	"github.com/renproject/pack"
+)
+
+var _ = Describe("Compat V0", func() {
+	It("should convert a QueryState response into a QueryShards response", func() {
+
+		shardsResponse, err := v0.ShardsResponseFromState(testutils.MockQueryStateResponse())
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(shardsResponse.Shards[0].Gateways[0].PubKey).Should(Equal("Akwn5WEMcB2Ff_E0ZOoVks9uZRvG_eFD99AysymOc5fm"))
+	})
+
+	It("should convert a v0 ParamsSubmitTx into a v1 ParamsSubmitTx", func() {
+		mr, err := miniredis.Run()
+		if err != nil {
+			panic(err)
+		}
+
+		client := redis.NewClient(&redis.Options{
+			Addr: mr.Addr(),
+		})
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		params := testutils.MockParamSubmitTxV0()
+
+		utxo := params.Tx.In.Get("utxo").Value.(v0.ExtBtcCompatUTXO)
+		vout := utxo.VOut.Int.String()
+		btcTxHash := utxo.TxHash
+		key := fmt.Sprintf("amount_%s_%s", btcTxHash, vout)
+		client.Set(key, 200000, 0)
+
+		bindingsOpts := txenginebindings.DefaultOptions().
+			WithNetwork("testnet")
+
+		bindingsOpts.WithChainOptions(multichain.Bitcoin, txenginebindings.ChainOptions{
```

Comment:
I think the correct usage should be 
```golang
    bindingsOpts = bindingsOpts.WithChainOptions(multichain.Bitcoin, txenginebindings.ChainOptions{}).
                                                       WithChainOptions(multichain.Bitcoin, txenginebindings.ChainOptions{})
                               
```

Reply:
Is the usage in `resolver/resolver_test.go` also incorrect then? I know that the `WithChainOptions` call has an effect, so it must be stateful and not just a builder function.